chia/server/server.py

```diff
@@ -493,12 +495,11 @@ def connection_closed(self, connection: WSChiaConnection, ban_time: int):
                 f"Invalid connection type for connection {connection.peer_host},"
                 f" while closing. Handshake never finished."
             )
+        self.cancel_tasks_from_peer(connection.peer_node_id)
```

Comment:
was there some race going on here?

Reply:
(it got moved before on disconnect)

Reply:
No i just did that as a precaution, i was scared the stuff below it might throw a exception